### ISSUE TITLE
JS implementation available?

### ISSUE THREAD

**Issue by tomyo** [jgm/djot#53]
Hi!
I like a lot the way djot language is defined, and I would like to try it out in web projects.

Is there a JS implementation? 
If not, i'm happy to collaborate => suggestions welcome.

Thanks!

**Comment by jgm** [comment]
No, there isn't, just the Lua.  It should be possible to write a JS implementation that closely follows the Lua implementation, so if you'd like to try this, be my guest!

Some notes on where I think the difficulties would lie:

- the Lua implementation operates with *byte* indices into the input string (because it's faster that way, and Lua uses bytestrings).  In JS it would be easier to operate with *character*  indices.
- I think there are tools for automatically translating Lua -> JS, but the tool I'm familiar with doesn't translate use of Lua pattern syntax, which is used heavily in this implementation.

**Comment by mandolyte** [comment]
Perhaps this should be moved over to discussions. I could probably donate a few hours (2-4h) on this. I've also searched for Lua to JS tools; many seemed to be abandoned. But this could be a first step - investigate what is available.
